diff --git a/dist/keystatic-core-api-generic.js b/dist/keystatic-core-api-generic.js
index aabdb1a49688d48a681ca4e06fb68ee7ca1e0869..618c1077d81181cd6150242284951349b61a276f 100644
--- a/dist/keystatic-core-api-generic.js
+++ b/dist/keystatic-core-api-generic.js
@@ -225,6 +225,8 @@ async function githubOauthCallback(req, config) {
     }
   });
   if (!tokenRes.ok) {
+    const errorText = await tokenRes.text();
+    console.error("GITHUB TOKEN EXCHANGE ERROR (!ok):", errorText);
     return {
       status: 401,
       body: 'Authorization failed'
@@ -235,10 +237,21 @@ async function githubOauthCallback(req, config) {
   try {
     tokenData = tokenDataResultType.create(_tokenData);
   } catch {
-    return {
-      status: 401,
-      body: 'Authorization failed'
-    };
+    console.error("GITHUB TOKEN EXCHANGE ERROR:", _tokenData);
+    if (_tokenData.access_token) {
+      tokenData = {
+        access_token: _tokenData.access_token,
+        expires_in: 31536000,
+        refresh_token: 'not-needed',
+        refresh_token_expires_in: 31536000,
+        scope: _tokenData.scope || ''
+      };
+    } else {
+      return {
+        status: 401,
+        body: 'Authorization failed'
+      };
+    }
   }
   const headers = await getTokenCookies(tokenData, config);
   if (state === 'close') {
diff --git a/dist/keystatic-core-api-generic.node.js b/dist/keystatic-core-api-generic.node.js
index b4c82e2d82fcb7fb3fa05835047d92a88866a608..eecf15d14b6febcf91d1168ea343b50337348418 100644
--- a/dist/keystatic-core-api-generic.node.js
+++ b/dist/keystatic-core-api-generic.node.js
@@ -428,6 +428,8 @@ async function githubOauthCallback(req, config) {
     }
   });
   if (!tokenRes.ok) {
+    const errorText = await tokenRes.text();
+    console.error("GITHUB TOKEN EXCHANGE ERROR (!ok):", errorText);
     return {
       status: 401,
       body: 'Authorization failed'
@@ -438,10 +440,21 @@ async function githubOauthCallback(req, config) {
   try {
     tokenData = tokenDataResultType.create(_tokenData);
   } catch {
-    return {
-      status: 401,
-      body: 'Authorization failed'
-    };
+    console.error("GITHUB TOKEN EXCHANGE ERROR:", _tokenData);
+    if (_tokenData.access_token) {
+      tokenData = {
+        access_token: _tokenData.access_token,
+        expires_in: 31536000,
+        refresh_token: 'not-needed',
+        refresh_token_expires_in: 31536000,
+        scope: _tokenData.scope || ''
+      };
+    } else {
+      return {
+        status: 401,
+        body: 'Authorization failed'
+      };
+    }
   }
   const headers = await getTokenCookies(tokenData, config);
   if (state === 'close') {
